gnc/navigator_perception/scan_the_code_detector.cc

```diff
@@ -8,105 +8,114 @@ using namespace cv;
 ///////////////////////////////////////////////////////////////////////////////////////////////////
 
 ScanTheCodeDetector::ScanTheCodeDetector()
-  try : image_transport(nh),   rviz("/scan_the_code/visualization/detection") {
+try :
+    image_transport(nh),   rviz("/scan_the_code/visualization/detection")
+    {
 
-    stringstream log_msg;
-    init_ros(log_msg);
+        stringstream log_msg;
+        init_ros(log_msg);
 
-    log_msg << "\nInitializing ScanTheCodeDetector:\n";
-    int tab_sz = 4;
+        log_msg << "\nInitializing ScanTheCodeDetector:\n";
+        int tab_sz = 4;
 
-    PerceptionModel model = PerceptionModel(model_width, model_height);
-    model_fitter = new StereoModelFitter(model, debug_image_pub);
+        PerceptionModel model = PerceptionModel(.1905,.381,4);
+        model_fitter = new StereoModelFitter(model, debug_image_pub);
+        // Start main detector loop
+        run_id = 0;
 
-    // Start main detector loop
-    run_id = 0;
-    boost::thread main_loop_thread(boost::bind(&ScanTheCodeDetector::run, this));
-    main_loop_thread.detach();
-    log_msg << setw(1 * tab_sz) << "" << "Running main detector loop in a background thread\n";
+        boost::thread main_loop_thread(boost::bind(&ScanTheCodeDetector::run, this));
+        main_loop_thread.detach();
+        log_msg << setw(1 * tab_sz) << "" << "Running main detector loop in a background thread\n";
 
-    log_msg << "ScanTheCodeDetector Initialized\n";
-    ROS_INFO(log_msg.str().c_str());
+        log_msg << "ScanTheCodeDetector Initialized\n";
+        ROS_INFO(log_msg.str().c_str());
 
-    } catch (const exception &e) {
+    }
+catch (const exception &e)
+    {
         ROS_ERROR("Exception from within ScanTheCodeDetector constructor "
-            "initializer list: ");
+                  "initializer list: ");
         ROS_ERROR(e.what());
     }
 
-ScanTheCodeDetector::~ScanTheCodeDetector() {
-  ROS_INFO("Killed Torpedo Board Detector");
+ScanTheCodeDetector::~ScanTheCodeDetector()
+{
+    ROS_INFO("Killed Torpedo Board Detector");
 }
 
 void ScanTheCodeDetector::validate_frame(Mat& current_image_left, Mat& current_image_right,
-                                          Mat& processing_size_image_left, Mat& processing_size_image_right
-                                          ){
-
+        Mat& processing_size_image_left, Mat& processing_size_image_right
+                                           )
+{
     // Prevent segfault if service is called before we get valid img_msg_ptr's
-    if (model_fitter->left_most_recent.image_msg_ptr == NULL || model_fitter->right_most_recent.image_msg_ptr == NULL) {
-      ROS_WARN("Torpedo Board Detector: Image Pointers are NULL.");
-      throw "ROS ERROR";
-    }
-    double sync_thresh = 0.5;
+    if (left_most_recent.image_msg_ptr == NULL || right_most_recent.image_msg_ptr == NULL)
+        {
+            throw "Torpedo Board Detector: Image Pointers are NULL.";
```

Comment:
Do you have code catching this? This condition should not crash the program, just skip the frame processing for this iteration.